draft-ietf-httpbis-message-signatures.md

```diff
@@ -289,6 +289,18 @@ NOTE: '\' line wrapping per RFC 8792
 
 Note: these are shown here using the line wrapping algorithm in {{RFC8792}} due to limitations in the document format that strips trailing spaces from diagrams.
 
+Any HTTP field content identifiers MAY have the following parameters in specific circumstances.
```

Comment:
First, content identifiers -> component identifiers.

I think this paragraph is over-generalizing. This would only make sense if Structured Fields were a recursive structure.

Agreed that the `absent` modifier can apply to any identifier (my misgivings about it aside).

I'm less sure about `sf`: can it be applied meaningfully to query parameters?

And `key` is very specific to dictionaries, so it can only apply to full headers.

The last one is `req` that was discussed elsewhere, that one could apply to all component identifiers.

Reply:
This paragraph is very specifically talking about HTTP field component identifiers, not all component identifiers. Query parameters are a derived component, not a field.

Reply:
Fair enough. But the paragraph still leaves open the question whether you can have both `key` and `absent` in the same field component identifier, i.e. if you can signal the intent to **not** include a particular dictionary member.